<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html><head>
  
  <meta content="text/html; charset=ISO-8859-1" http-equiv="content-type">
  <title>Tutorial on Phylogenetics &amp; Rapidly Evolving Pathogens</title>

  </head><body>
<h2>Tutorial on Phylogeny and Rapidly Evolving Pathogens</h2>


<br>


Handouts and supporting material for the <a href="http://dimacs.rutgers.edu/Workshops/PhyloTutorial/">DIMACS Tutorial on Phylogenetic Trees and Rapidly Evolving Pathogens</a>,<br>


June 19 - 20, 2006, DIMACS Center, CoRE Building, Rutgers University.<br>


<br>



<ul>


  <li><a href="dimacsOverview20060619.pdf">Overview</a>, Katherine St. John</li>
  <li><a href="DIMACS06TB.ppt">Parsimony Reconstruction of Phylogenetic Trees</a>, Trevor Bruen</li>
  <li><a href="DIMACS06RB.pdf">Using Maximum Likelihood for Phylogenetic Tree Reconstruction</a>, Rachel Bevan<br>
  </li>
  <li>Hands On Session 1:&nbsp; Constructing Trees<br clear="all">
    <ul>
      <li><a href="dimacsHandsOn20060619.pdf">Slides</a></li>
      <li><a href="murphy.nex">Sequences</a> (PAUP format)</li>
      <li><a href="murphy.phy">Sequences</a> (Phylip format)</li>
      <li><a href="murphy.fa">Sequences</a> (FASTA format)</li>
      <li><a href="shortSequences.nex">Small set of sequences</a> (just ADORA gene, PAUP format) <br clear="all">
&nbsp;  </li>
    </ul>
  </li>
  <li><a href="CoalescentAndMEP.ppt">The Coalescent and Measurably Evolving Populations</a>, Alexis Drummond</li><li><a href="Mike_Steel.ppt">Tricks for trees: Having reconstructed phylogenies what can we do with them?</a>, Mike Steel<br>
  </li>
  <li>&nbsp;Hands On Session 2: Getting Data from On-line Sources <br clear="all">
    <ul>
      <li><a href="dimacsHandsOn20060620.pdf">Slides</a></li>
      <li><a href="alignedHIV.phy">Aligned Sequences</a> (phylip format)</li>
      <li><a href="alignedHIV.dis.dist">Distance Matrix</a><br>
      </li>
      <li><a href="alignedHIV.tree">BioNJ Tree</a></li>
    </ul>
  </li>

</ul>


<br>


<br>

</body></html>