<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"><html><head><title>ideas</title><meta http-equiv="content-type" content="text/html; charset=ISO-8859-1"></head><body><h1>Project Ideas</h1><h2> CMP 464/788: Introduction to Computational Biology<br>Lehman College, City University of New York<br>Fall 2003</h2><br><h4>Sequence Alignment</h4><ul>  <li>Present the analysis of running times (ie proofs) for the standarddynamic programming algorithms for pairwise alignment (Needleman-Wunsch,Smith-Waterman)</li>  <li>Present the concept of Hidden Markov Models (HMMs) and how they areused in sequence alignment. &nbsp;Compare and constrast various methods thatuse HMMs.</li>  <li>How are FASTA and BLAST different? &nbsp;Explain and compare the strengths and limitations of these two approaches.<br>  </li></ul><br><h4><br>Phylogeny</h4><ul>  <li><br>  </li></ul><br></body></html>