<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"><html><head><title>CMP 464/788, Lehman College, CUNY Fall 2003</title></head><body bgcolor="#ffffff" link="#003399" vlink="#236b8e"> <h2>Computer Science 464/788:&nbsp;</h2><h2>Introduction to Computational Biology<br>Lehman College, City University of New York<br>Fall 2003</h2><hr><b>Instructor:</b> Prof. Katherine St. John<br><b>E-mail:</b> stjohn@lehman.cuny.edu<br><b>Phone:</b> 718-960-7423<br><b>Office:</b> G 137D<br><b>Office Hours:</b> Tuesdays 3-4:30, 6-7:30, and by appointment.<br><b>Lecture:</b> Tuesdays and Thursdays 7:45-9:25pm in G 225.<br><hr><h3>Announcements:</h3><ul>     <li>Some students cheated on the initial parts of the project.&nbsp; As such, each project presentation is being double checked<br>for plagiarism.&nbsp; This is extremely time-consuming, and so, the gradesfor all presentations will not be ready until next week. &nbsp;Sorry forthe delay! &nbsp;As each presentation is checked, it's grade will be posted.&nbsp;So, some students already have their grades; others will be postedsoon.<br></li><li>The <a href="Presentation_Schedule.html">presentation schedule</a> for the <a href="project.html">projects</a> is now available, as the <a href="projectQuizzes.html">quizzes</a> based on the projects and <a href="../../photos/compBioF03/compBioF03.html">photos</a> from the poster session.<br>  </li><li>This course is being offered at the undergraduatelevel (CMP 464) and masters level (CMP 788). &nbsp;The lectures for bothcourses will be the same. &nbsp;Those taking the course for graduate creditwill have extra reading and programming assignments (in addition to thosefor the undergraduates), will have quizzes that test the understanding of thematerial at a deeper level, and will give a presentation to the class. <br>  </li></ul><h3>Handouts:</h3><ul><li> <a href="syl.html">Syllabus</a></li>    <li><a href="outline.html">Outline</a></li><li><font>"<a href="http://kim.bio.upenn.edu/%7Ejkim/media/YaleScienceJournal.pdf">Computers Are from Mars, Organisms are from Venus</a>" by J. Kim, 2002.</font></li>  <li>Handouts for lecture on Tuesday, 9 September:<br clear="all">    <ul>      <li>"<a href="http://www.blc.arizona.edu/Molecular_Graphics/DNA_Structure/DNA_Tutorial.HTML">Introduction to DNA Structure</a>" by Prof. Richard B. Hallick, U. Arizona. &nbsp;</li>      <li>"<a href="http://allserv.rug.ac.be/%7Eavierstr/principles/seq.html">Principle of Sequencing</a>" by Dr. Andy Vierstraete, U. Ghent.</li>      <li>Pairwise Alignment: "<a href="http://www.sbc.su.se/%7Eper/molbioinfo2001/dynprog/dynamic.html">Dynamic Programming Example</a>" &amp; "<a href="http://www.sbc.su.se/%7Eper/molbioinfo2001/dynprog/adv_dynamic.html">Advanced DP Example</a>" by Dr. Eric C. Rouchka, Washington U., &nbsp;St. Louis.<br>      </li>    </ul>  </li>  <li><a href="quiz1.html">Quiz 1</a> (due 18 September)</li>  <li>Handouts for lecture on Tuesday, 23 September:<br clear="all">    <ul>      <li>Background on Dynamic Programming:<br clear="all">        <ul>          <li><a href="http://www.nist.gov/dads/HTML/dynamicprog.html">Definition</a> from the National Institute of Standards and Technology (NIST</li>          <li>"<a href="http://plus.maths.org/issue3/dynamic/">Dynamic Programming: An Introduction</a>"by David K. Smith and PASS Maths (Part of the Millennium Mathematics Project sponsored by the Univeristy of Cambridge).</li>          <li><a href="http://www.csse.monash.edu.au/%7Elloyd/tildeAlgDS/Dynamic/">Definition</a> and <a href="http://www.csse.monash.edu.au/%7Elloyd/tildeAlgDS/Dynamic/Edit/">Edit Distance example</a> by Dr. Lloyd Allison, Monash U.</li>        </ul>      </li>      <li><a href="http://www.molbiol.ox.ac.uk/help/scoringmatrices.htm">Scoring Matrices</a> from Oxford University Bioinfomatics Centre.<br>      </li>    </ul>  </li>  <li><a href="quiz2.html">Quiz 2</a> (due 25 September)</li><li>Handouts for lecture on Thursday, 2 October:<br clear="all">    <ul>      <li>Chapter 3 &amp; 4 <a href="http://examples.oreilly.com/begperlbio/">Examples</a> from <a href="http://www.oreilly.com/catalog/begperlbio/">Beginning Perl for Bioinformatics</a> by James Tisdall</li></ul>      <ul>        </ul>  </li>  <li><a href="quiz3.html">Quiz 3</a> (due 2 October)</li><li>Handouts for lecture on Thursday, 9 October:<br clear="all">    <ul>      <li><a href="introPhyloNotes.pdf">Introduction to Phylogeny</a> Lecture Notes&nbsp;</li>      <li>Chapter 5 <a href="http://examples.oreilly.com/begperlbio/">Examples</a> (loop and decision statements) from <a href="http://www.oreilly.com/catalog/begperlbio/">Beginning Perl for Bioinformatics</a> by James Tisdall</li>    </ul>  </li>  <li>Handouts for lecture on Tuesday, 14 October:<br clear="all">    <ul><li><a href="project.html">Project</a> Information<br>      </li>      <li>Multiple Sequence Alignment:<br clear="all">        <ul>          <li><a href="http://www.hgmp.mrc.ac.uk/embnet.news/vol2_1/align.html">Overview of Multiple Sequence Alignment</a> by Dr. Guy Bottu for embnet.news.<br>          </li>          <li><a href="http://www.dkfz-heidelberg.de/tbi/bioinfo/Phylogeny/Iterative/">Iterative Alignment Strategy</a> &nbsp;by Dr. Hannes Luz, German Cancer Research Center and the Max Planc Institute for Molecular Genetics.</li>        </ul>      </li><li>Using software: <br clear="all">        <ul>          <li><a href="http://www.ncbi.nlm.nih.gov/Education/BLASTinfo/information3.html">BLAST</a> tutorial (pairwise sequence alignment)<br>          </li>          <li><a href="http://www.ebi.ac.uk/fasta33/fasta33_frame.html">Fasta</a> overview and tutorial (pairwise sequence alignment)<br>          </li>          <li><a href="http://www.ebi.ac.uk/clustalw/clustalw_frame.html">Clustalw</a> overview and tutorial (multiple sequence alignment)<br>          </li>        </ul>      </li><li>Chapter 5 <a href="http://examples.oreilly.com/begperlbio/">Examples</a> (more on regular expressions and loops) from <a href="http://www.oreilly.com/catalog/begperlbio/">Beginning Perl for Bioinformatics</a> by James Tisdall</li></ul>  </li>  <li>Handouts for lecture on Thursday, 16 October:<br clear="all">    <ul><li><a href="http://www-ab.informatik.uni-tuebingen.de/teaching/ws02/script/phylogeny.ps.gz">Phylogenetic Tree Notes</a>         (<a href="danielNotes/phylogeny.pdf">pdf version</a>) from <font><a href="http://www-ab.informatik.uni-tuebingen.de/people/huson/welcome_en.html">Prof. Daniel Huson,</a></font>        University of Tuebingen, Germany.      </li><li>Chapter 5 <a href="http://examples.oreilly.com/begperlbio/">Examples</a> (subroutines) from <a href="http://www.oreilly.com/catalog/begperlbio/">Beginning Perl for Bioinformatics</a> by James Tisdall</li></ul>  </li>  <li>Handouts for lecture on Tuesday, 21 October:<br clear="all">    <ul><li><a href="http://www-ab.informatik.uni-tuebingen.de/teaching/ws02/script/phylogeny.ps.gz">Phylogenetic Tree Notes</a>         (<a href="file:///smore/Users/stjohn/public_html/teaching/compbio/danielNotes/phylogeny.pdf">pdf version</a>)<font>and <a href="http://www-ab.informatik.uni-tuebingen.de/teaching/ws02/script/splits.ps.gz">Phylogenetic Networks Notes </a></font>(<a href="danielNotes/splits.pdf">pdf version</a>) from <font><a href="http://www-ab.informatik.uni-tuebingen.de/people/huson/welcome_en.html">Prof. Daniel Huson,</a></font>        University of Tuebingen, Germany.</li><li>Chapter 6 <a href="http://examples.oreilly.com/begperlbio/">Examples</a> (command-line arguments &amp; passing variables) from <a href="http://www.oreilly.com/catalog/begperlbio/">Beginning Perl for Bioinformatics</a> by James Tisdall</li></ul>  </li>  <li><a href="quiz4.html">Quiz 4</a> (due 21 October)</li><li>Handouts for lecture on Thursday, 23 October:<br clear="all">    <ul><li><a href="http://www-ab.informatik.uni-tuebingen.de/teaching/ws02/script/phylogeny.ps.gz">Phylogenetic Tree Notes</a>         (<a href="file:///smore/Users/stjohn/public_html/teaching/compbio/danielNotes/phylogeny.pdf">pdf version</a>)<font>and <a href="http://www-ab.informatik.uni-tuebingen.de/teaching/ws02/script/splits.ps.gz">Phylogenetic Networks Notes </a></font>(<a href="file:///smore/Users/stjohn/public_html/teaching/compbio/danielNotes/splits.pdf">pdf version</a>) from <font><a href="http://www-ab.informatik.uni-tuebingen.de/people/huson/welcome_en.html">Prof. Daniel Huson,</a></font>        University of Tuebingen, Germany.</li><li><a href="file:///smore/Users/stjohn/public_html/teaching/compbio/classNotes231003.pdf">Lecture notes</a> on consensus methods,<br>      </li><li>Chapter 7 <a href="http://examples.oreilly.com/begperlbio/">Examples</a> (subroutines &amp; modules) from <a href="http://www.oreilly.com/catalog/begperlbio/">Beginning Perl for Bioinformatics</a> by James Tisdall.</li></ul>  </li>  <li>Handouts for lecture on Tuesday, 28 October:<br clear="all">    <ul><li><a href="http://www-ab.informatik.uni-tuebingen.de/teaching/ws02/script/phylogeny.ps.gz">Phylogenetic Tree Notes</a>         (<a href="file:///smore/Users/stjohn/public_html/teaching/compbio/danielNotes/phylogeny.pdf">pdf version</a>)<font>and <a href="http://www-ab.informatik.uni-tuebingen.de/teaching/ws02/script/splits.ps.gz">Phylogenetic Networks Notes </a></font>(<a href="file:///smore/Users/stjohn/public_html/teaching/compbio/danielNotes/splits.pdf">pdf version</a>) from <font><a href="http://www-ab.informatik.uni-tuebingen.de/people/huson/welcome_en.html">Prof. Daniel Huson,</a></font>        University of Tuebingen, Germany.</li><li><a href="classNotes281003.pdf">Lecture notes</a> on networks and comparing trees,<br>      </li>      <li>"<a href="http://www.cs.utexas.edu/users/amenta/pubs/treeviz.pdf">Case Study: Visualizing Sets of Evolutionary Trees</a>" Amenta &amp; Klingner, 2002,<br>        <font> </font></li><li><font><a href="http://www.cs.ubc.ca/%7Ejslack/imager-web/Research/2003/TreeJuxtaposer/">TreeJuxtaposer.</a></font>           by Prof. Tamara Munzner, University of British Columbia, Canada      </li><li>Chapter 7 <a href="http://examples.oreilly.com/begperlbio/">Examples</a> (random number generation) from <a href="http://www.oreilly.com/catalog/begperlbio/">Beginning Perl for Bioinformatics</a> by James Tisdall.</li></ul>  </li>  <li><a href="program1.html">Programming Assignment 1</a> (target date of 28 October)</li>  <li>Handouts for lecture on Thursday, 30 October:<br clear="all">    <ul><li><a href="http://www-ab.informatik.uni-tuebingen.de/teaching/ws02/script/phylogeny.ps.gz">Phylogenetic Tree Notes</a>         (<a href="file:///smore/Users/stjohn/public_html/teaching/compbio/danielNotes/phylogeny.pdf">pdf version</a>)<font>and <a href="http://www-ab.informatik.uni-tuebingen.de/teaching/ws02/script/splits.ps.gz">Phylogenetic Networks Notes </a></font>(<a href="file:///smore/Users/stjohn/public_html/teaching/compbio/danielNotes/splits.pdf">pdf version</a>) from <font><a href="http://www-ab.informatik.uni-tuebingen.de/people/huson/welcome_en.html">Prof. Daniel Huson,</a></font>        University of Tuebingen, Germany.</li><li><a href="file:///smore/Users/stjohn/public_html/teaching/compbio/classNotes301003.pdf">Lecture notes</a> on testing algorithm empirically.</li><li>Paperby Prof. Moret,&nbsp;&nbsp; "Towards a discipline of experimental algorithmics,"&nbsp;&nbsp;in Data Structures, Near Neighbor Searches, and Methodology:&nbsp;&nbsp;Fifth and Sixth DIMACS Implementation Challenges, M.H. Goldwasser, D.S. Johnson,and C.C. McGeoch, eds, DIMACS Monographs 59 , 197-213,&nbsp;&nbsp; AmericanMathematical Society, Providence, 2002. Available from <a href="http://www.cs.unm.edu/%7Emoret/papers.html">http://www.cs.unm.edu/~moret/papers.html</a>.<br>      </li><li>Chapter 7 <a href="http://examples.oreilly.com/begperlbio/">Examples</a> (random number generation) from <a href="http://www.oreilly.com/catalog/begperlbio/">Beginning Perl for Bioinformatics</a> by James Tisdall.</li></ul>  </li><li><a href="file:///smore/Users/stjohn/public_html/teaching/compbio/quiz5.html">Quiz 5</a> (due 30 October)</li><li>Handouts for lecture on Tuesday, 4 November:<br clear="all">    <ul><li><a href="http://www-ab.informatik.uni-tuebingen.de/teaching/ws02/script/protstruct2d.ps.gz">Protein Secondary Structure</a>&nbsp;         (<a href="danielNotes/protstruct2d.pdf">pdf version</a>) from <font><a href="http://www-ab.informatik.uni-tuebingen.de/people/huson/welcome_en.html">Prof. Daniel Huson,</a></font>        University of Tuebingen, Germany.</li><li><a href="http://www.stanford.edu/group/pandegroup/folding/education/">Overview</a>from Folding@Home (we'll discuss the overview plus background info (see sidebar)on amino acids, proteins, diseases, and molecular modeling)<br>      </li><li>Chapter 8 <a href="http://examples.oreilly.com/begperlbio/">Examples</a> (hashes) from <a href="http://www.oreilly.com/catalog/begperlbio/">Beginning Perl for Bioinformatics</a> by James Tisdall.</li></ul>  </li>  <li>Handouts for lecture on Tuesday, 11 November:<br clear="all">    <ul><li><a href="http://www-ab.informatik.uni-tuebingen.de/teaching/ws02/script/protstruct3d.ps.gz">Protein Tertiary Structure</a>&nbsp;         (<a href="danielNotes/protstruct3d.pdf">pdf version</a>) from <font><a href="http://www-ab.informatik.uni-tuebingen.de/people/huson/welcome_en.html">Prof. Daniel Huson,</a></font>        University of Tuebingen, Germany.</li><li>Chapter 8 <a href="http://examples.oreilly.com/begperlbio/">Examples</a> (more on hashes, genetic code, and FASTA format) from <a href="http://www.oreilly.com/catalog/begperlbio/">Beginning Perl for Bioinformatics</a> by James Tisdall.</li></ul>  </li>    <li><a href="program2.html">Programming Assignment 2</a> (target date of 11 November)</li><li>Handouts for lecture on Thursday, 13 November:<br clear="all">    <ul><li><a href="http://www-ab.informatik.uni-tuebingen.de/teaching/ws02/script/arrays.ps.gz">DNA Arrays</a>&nbsp;         (<a href="danielNotes/arrays.pdf">pdf version</a>) from <font><a href="http://www-ab.informatik.uni-tuebingen.de/people/huson/welcome_en.html">Prof. Daniel Huson,</a></font>        University of Tuebingen, Germany.</li><li><a href="http://www.microarrays.org/">microarray.com</a>: &nbsp;a source for information about microarray technology.</li>      <li>NCBI's <a href="http://www.ncbi.nlm.nih.gov/About/primer/microarrays.html">primer on microarrays</a>.</li><li>Chapter 10 <a href="http://examples.oreilly.com/begperlbio/">Examples</a> (Protein Data Bank) from <a href="http://www.oreilly.com/catalog/begperlbio/">Beginning Perl for Bioinformatics</a> by James Tisdall.</li></ul>  </li>  <li><a href="quiz6.html">Quiz 6</a> (due 13 November)</li>  <li>Handouts for lecture on Tuesday, 18 November:<br clear="all">    <ul><li><a href="http://www-ab.informatik.uni-tuebingen.de/teaching/ws02/script/arrays.ps.gz">DNA Arrays</a>&nbsp;         (<a href="danielNotes/arrays.pdf">pdf version</a>) from <font><a href="http://www-ab.informatik.uni-tuebingen.de/people/huson/welcome_en.html">Prof. Daniel Huson,</a></font>        University of Tuebingen, Germany.</li><li><a href="http://cellcycle-www.stanford.edu/">Yeast cell cycle analysis project</a> via microarrays from Stanford University.</li>      <li><a href="http://www.ncbi.nlm.nih.gov/About/primer/snps.html">An Introduction to SNPs</a> from the National Center for Biotechnology Information.</li>      <li><a href="http://www.genome.gov/10001665">Haplotype Mapping Project Report</a> from the National Human Genome Research Institute.<br>      </li><li>Chapter 12 <a href="http://examples.oreilly.com/begperlbio/">Examples</a> (BLAST) from <a href="http://www.oreilly.com/catalog/begperlbio/">Beginning Perl for Bioinformatics</a> by James Tisdall.</li></ul>  </li>  <li>Handouts for lecture on Thursday, 20 November:<br clear="all">    <ul><li><a href="http://www-ab.informatik.uni-tuebingen.de/teaching/ws02/script/assembly.ps.gz">Genome Assembly</a>&nbsp;         (<a href="danielNotes/assembly.pdf">pdf version</a>) from <font><a href="http://www-ab.informatik.uni-tuebingen.de/people/huson/welcome_en.html">Prof. Daniel Huson,</a></font>        University of Tuebingen, Germany.</li><li><a href="http://www.ncbi.nlm.nih.gov/About/primer/mapping.html">An Introduction to Genome Mapping</a> from the National Center for Biotechnology Information.<br>      </li><li>Chapter 9 <a href="http://examples.oreilly.com/begperlbio/">Examples</a> (restriction maps &amp; regular expressions) from <a href="http://www.oreilly.com/catalog/begperlbio/">Beginning Perl for Bioinformatics</a> by James Tisdall.</li></ul>  </li>    <li><a href="program3.html">Programming Assignment 3</a> (target date of 25 November)<br>  </li></ul><h3>Useful Links:</h3><ul><li>Textbook: <a href="http://www.oreilly.com/catalog/bioskills/index.html">http://www.oreilly.com/catalog/bioskills/index.html</a></li><li>Blackboard: <a href="http://blackboard.lehman.cuny.edu">http://blackboard.lehman.cuny.edu</a><br>  </li></ul><br><br><br><br><br><br><br><br><br></body></html>