<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"><html><head><title>Outline for CMP 464/788, Lehman College, Fall 2003</title></head><body bgcolor="#ffffff" link="#003399" vlink="#236b8e"><h2> CMP 464/788: Introduction to Computational Biology<br>Lehman College, City University of New York<br>Fall 2003</h2><hr>Note:  This outline is tenative and subject to change.<hr><table border="1" width="100%" nosave="">	<tbody><tr nosave="">	<td nosave="">		<b><font>Date/Time:</font></b></td>				<td><b><font>Description:</font></b></td>		<td><b><font>Reading:</font></b></td>		<td><b><font>Overheads:</font></b></td>	</tr>	<tr> <td><b><font>2 September<br>7:45-9:25pm<br></font></b></td>				<td><font><a href="syl.html">First Day Details</a>, Overview of Topics, What is Computational Biology?</font></td>		<td><font>Chapters 1-2,<br>"<a href="http://kim.bio.upenn.edu/%7Ejkim/media/YaleScienceJournal.pdf">Computers Are from Mars, Organisms are from Venus</a>," Kim 2002 <br>      </font></td>		<td>		    <a href="acadPol.pdf">Academic Policy</a><br>		    <br></td>		</tr><tr> <td><b><font>4&nbsp;</font></b><b><font>September<br>7:45-<font color="#ff0000">9:45</font>pm<br></font></b></td>				<td>Core Concepts of &nbsp;Molecular Biology, Computational Approaches to Biological Questions, Computational Biology Experiments.<br></td>		<td>Chapter 2<br></td>		<td><br></td>	</tr><tr> <td><b><font>9&nbsp;</font></b><b><font>September<br></font></b><b><font>7:45-<font color="#ff0000">9:45</font>pm</font></b><br></td>				<td><font> 			</font>Sequence Alignment: &nbsp;Biological Background &amp; Pairwise Sequence Alignment Algorithms, Dynamic Programming. <br></td>		<td>Chapters 2, 7<br></td>		<td><a href="http://www.blc.arizona.edu/Molecular_Graphics/DNA_Structure/DNA_Tutorial.HTML">Introduction to DNA Structure</a>,&nbsp;<a href="http://allserv.rug.ac.be/%7Eavierstr/principles/seq.html">Principle of Sequencing</a>,&nbsp;<a href="http://www.sbc.su.se/%7Eper/molbioinfo2001/dynprog/dynamic.html">Dynamic Programming &amp; Alignment</a></td>		</tr><tr> <td><b><font>11 </font></b><b><font>September<br></font></b><b><font>7:45-<font color="#ff0000">9:45</font>pm</font></b><br></td>				<td>Bioinformatics Workstation Basics: Files &amp; Directories in Unix, Unix Shell, <font>Filtering Data.</font><br><font>			</font></td>		<td>Chapters 3-5<br></td>		<td><br></td>	</tr><tr> <td><b><font>16 </font></b><b><font>September</font></b><br></td>				<td>      <font color="#ff0000">No Class</font> (Prof. St. John at <a href="http://www.conferences.hu/ALGO2003/algo_2003.htm">WABI</a>)<br></td>		<td><br></td>		<td><br></td> 	</tr><tr> <td><b><font>18 </font></b><b><font>September</font></b><b><font><br></font></b></td>				<td><font color="#ff0000">No Class</font> (Prof. St. John at <a href="http://www.conferences.hu/ALGO2003/algo_2003.htm">WABI</a>).<br>      <font><a href="file:///smore/Users/stjohn/public_html/teaching/compbio/quiz1.html">Quiz 1</a></font><font>&nbsp;</font><font> (Using Blackboard, Academic Policy) due.</font><br><font>		</font></td>		<td><font>&nbsp;</font></td>		<td><font>&nbsp;</font></td>	</tr><tr> <td><b><font>23 </font></b><b><font>September<br></font></b><b><font>7:45-<font color="#ff0000">9:45</font>pm</font></b><br></td>				<td><font>More on Pairwise Sequence Alignment: </font>Local vs. Global Alignment,&nbsp;<font> Aligning Proteins. </font><br></td>		<td>Background on DP (<a href="http://www.nist.gov/dads/HTML/dynamicprog.html">NIST Definition,</a> <a href="http://plus.maths.org/issue3/dynamic/">DP: Informal Introduction</a>, <a href="http://www.csse.monash.edu.au/%7Elloyd/tildeAlgDS/Dynamic/">Allison's Definition</a><a href="http://www.sbc.su.se/%7Eper/molbioinfo2001/dynprog/dynamic.html">, </a><a href="http://www.csse.monash.edu.au/%7Elloyd/tildeAlgDS/Dynamic/Edit/">Edit Distance example</a>), Chapter 7, <a href="http://www.molbiol.ox.ac.uk/help/scoringmatrices.htm">Scoring Matrices</a>.<br></td>		<td><a href="http://www.sbc.su.se/%7Eper/molbioinfo2001/dynprog/dynamic.html"><br>Dynamic Programming &amp; Alignment</a>,&nbsp;<a href="http://www.sbc.su.se/%7Eper/molbioinfo2001/dynprog/adv_dynamic.html">Advanced DP Example</a></td>	</tr><tr> <td><b><font>25 </font></b><b><font>September<br></font></b><b><font>7:45-<font color="#ff0000">9:45</font>pm</font></b><br></td>				<td><font>More on Unix shell, Regular expressions, Shell scripts, </font><font>Playing Nicely with Others in a Shared Environment</font>, Intro to Perl: string, pattern matching basics and printing.<br>      <font><a href="file:///smore/Users/stjohn/public_html/teaching/compbio/quiz2.html">Quiz 2</a></font><font>&nbsp; (Core Concepts of Molecular Biology) due.</font><font><br><font color="green">			</font>		</font></td>		<td><font>Chapters 4-5, 12</font></td>		<td><font>&nbsp;</font>  <br>      <a href="http://examples.oreilly.com/begperlbio/">Examples</a> from <a href="http://www.oreilly.com/catalog/begperlbio/">Beginning Perl for Bioinformatics</a> by James Tisdall.&nbsp;</td> 		</tr><tr> <td><b><font>30 </font></b><b><font>September</font></b><br></td>				<td><font color="#ff0000">No Class</font><font color="#ff0000"><font color="#000000">&nbsp;(Prof. St. John at <a href="http://www.inra.fr/eccb2003/">ECCB</a>)</font><br>      </font></td>		<td><br></td>		<td>		    <br></td>	</tr><tr> <td><b><font>2 October<br></font></b><b><font>7:45-<font color="#ff0000">9:45</font>pm</font></b><br></td>				<td><font>Biological Research on the Web: public databases</font>, on-line sources, Intro to Perl: files and arrays.<br>      <font>			</font><font><a href="file:///smore/Users/stjohn/public_html/teaching/compbio/quiz3.html">Quiz 3</a></font><font> (Sequence Alignment) due.</font><br></td>		<td><font>Chapters 6, 12</font></td>		<td>  <a href="http://examples.oreilly.com/begperlbio/">Examples</a> from <a href="http://www.oreilly.com/catalog/begperlbio/">Beginning Perl for Bioinformatics</a> by James Tisdall.&nbsp;<br>      <font>&nbsp;</font></td> 	</tr><tr> <td><b><font>7&nbsp;</font></b><b><font>October<br></font></b><br></td>				<td><font><font color="#ff0000">Classes Follow Monday Schedule: &nbsp;No Class</font> 		</font></td>		<td><br></td>		<td> <br></td>	</tr><tr> <td><b><font>9&nbsp;</font></b><b><font>October</font></b><b><font><br></font></b><b><font>7:45-<font color="#ff0000">9:45</font>pm<br></font></b></td>				<td>More on Multiple Sequence Alignment, Introductionto Phylogeny: Building an Evolutionary History, Intro to Perl: decisionsand loops.<br></td>		<td>Chatpters 8, 12<br></td>		<td><a href="introPhyloNotes.pdf">Introduction to Phylogeny</a> Lecture Notes, Sequence Alignment Software (see front page for links), <a href="http://examples.oreilly.com/begperlbio/">Examples</a> from <a href="http://www.oreilly.com/catalog/begperlbio/">Beginning Perl for Bioinformatics</a> by James Tisdall.&nbsp;<br></td>		</tr><tr> <td><b><font>14 </font></b><b><font>October<br></font></b><b><font>7:45-<font color="#ff0000">9:45</font>pm</font></b><br></td>				<td><font color="red">					</font>Information about&nbsp;<font>				<font color="#000000"><a href="file:///smore/Users/stjohn/public_html/teaching/compbio/project/index.html">Project</a></font></font> and Programming Assignments,<br>More on Multiple Sequence Alignment: Profiles &amp; Motifs, <br>Quick Overview of Alignment software <font>(blast, fasta, clustalw), </font>Intro to perl: finding motifs<br>      <font>			</font></td>		<td>Chapters 8, 12<br></td>		<td>Sequence Alignment Software (see front page for links),		    More on Phylogeny Lecture Notes, <br>      <a href="http://examples.oreilly.com/begperlbio/">Examples</a> from <a href="http://www.oreilly.com/catalog/begperlbio/">Beginning Perl for Bioinformatics</a> by James Tisdall.&nbsp;&nbsp;</td> 		</tr><tr> <td><b><font>16 </font></b><b><font>October<br></font></b><b><font>7:45-<font color="#ff0000">9:45</font>pm</font></b><br></td>				<td><a href="quiz5.html"><font>			</font></a>Building, Using, and Visualizing Phylogenetic Trees: Heuristics for the NP-hard search problems.<br>More on perl: Subroutines <br><font>				</font></td>		<td>Section 8.3 &amp; <a href="http://www-ab.informatik.uni-tuebingen.de/people/huson/welcome_en.html">Prof. Huson's</a> <a href="http://www-ab.informatik.uni-tuebingen.de/teaching/ws02/script/phylogeny.ps.gz">Phylogenetic Tree Notes</a><br><br></td>		<td><a href="http://examples.oreilly.com/begperlbio/">Examples</a> from <a href="http://www.oreilly.com/catalog/begperlbio/">Beginning Perl for Bioinformatics</a> by James Tisdall.&nbsp;<font>&nbsp;</font></td> 	</tr><tr> <td><b><font>21 </font></b><b><font>October<br></font></b><b><font>7:45-<font color="#ff0000">9:45</font>pm</font></b><br></td>				<td>More on Building, Using, and Visualizing Phylogenetic Trees. Phylogenetic Networks (for non-tree like evolution).<br>Gene Trees vs. Species Trees.<br>More on perl: passing variables.<br>      <font><a href="file:///smore/Users/stjohn/public_html/teaching/compbio/quiz4.html">Quiz 4</a> </font><font>(Using Unix)</font><font> due.			</font><br><font>			</font></td>		<td><font><a href="http://www-ab.informatik.uni-tuebingen.de/people/huson/welcome_en.html">Prof. Huson's</a> notes on <a href="http://www-ab.informatik.uni-tuebingen.de/teaching/ws02/script/phylogeny.ps.gz">Phylogenetic Trees</a> and <a href="http://www-ab.informatik.uni-tuebingen.de/teaching/ws02/script/splits.ps.gz">Networks</a>,</font><br></td>		<td><br>      <a href="http://examples.oreilly.com/begperlbio/">Examples</a> from <a href="http://www.oreilly.com/catalog/begperlbio/">Beginning Perl for Bioinformatics</a> by James Tisdall.&nbsp;</td>	</tr><tr> <td><b><font>23 </font></b><b><font>October </font></b><b><font><br>7:45-<font color="#ff0000">9:45</font>pm</font></b></td>				<td><font>				</font><a href="file:///smore/Users/stjohn/public_html/teaching/compbio/quiz5.html"><font>			</font></a>Visualizing Trees. Phylo Software: PAUP, Mesquite, SplitsTree, TreeJuxtaposer. <br>Gene Trees vs. Species Trees.<br>More on perl: command-line arguments<br>      <font>				<font color="#000000"><a href="file:///smore/Users/stjohn/public_html/teaching/compbio/project/index.html">Project:</a></font>			<font color="#009900">Target Date for Initial Plan.</font>		</font><br><font>		</font></td>		<td><font> <br>      </font>"<a href="http://www.cs.utexas.edu/users/amenta/pubs/treeviz.pdf">Case Study: Visualizing Sets of Evolutionary Trees</a>" Amenta&amp;Klingner 2002,<br>      <font> <a href="http://www.cs.ubc.ca/%7Ejslack/imager-web/Research/2003/TreeJuxtaposer/">TreeJuxtaposer.</a></font><br>      <br></td>		<td><font></font>Lecture notes on <a href="classNotes231003.pdf">phylogeny</a>,<br>      <a href="http://examples.oreilly.com/begperlbio/">Examples</a> from <a href="http://www.oreilly.com/catalog/begperlbio/">Beginning Perl for Bioinformatics</a> by James Tisdall.&nbsp;</td> 	</tr><tr> <td><b><font>28 </font></b><b><font>October<br></font></b><b><font>7:45-<font color="#ff0000">9:45</font>pm</font></b><br></td>				<td><font color="red">			</font> Comparing Trees &amp; Calculating Distances (RF, TBR, SPR). <br>Summarizing Large Sets of Trees (Strict Consensus, Majority Rule).<br>More on perl: modules &amp; libraries of subroutines. <br>Target date for<br>      <a href="program1.html">Programming Assignment 1</a>.<br><font>		</font></td>		<td>      <font><a href="http://www-ab.informatik.uni-tuebingen.de/people/huson/welcome_en.html">Prof. Huson's</a> notes on <a href="http://www-ab.informatik.uni-tuebingen.de/teaching/ws02/script/phylogeny.ps.gz">Phylogenetic Trees</a> and <a href="http://www-ab.informatik.uni-tuebingen.de/teaching/ws02/script/splits.ps.gz">Networks</a></font></td>		<td>Lecture notes on <a href="classNotes281003.pdf">phylogenetic networks</a>,<br>      <a href="http://examples.oreilly.com/begperlbio/">Examples</a> from <a href="http://www.oreilly.com/catalog/begperlbio/">Beginning Perl for Bioinformatics</a> by James Tisdall.&nbsp;</td> 	</tr><tr> <td><b><font>30 </font></b><b><font>October<br></font></b><b><font>7:45-<font color="#ff0000">9:45</font>pm</font></b><br></td>				<td>Computer Simulations and Algorithm Experiments<br>Case Study: &nbsp;Comparing Phylogenetic Tree Reconstruction Algorithms. <br>More on perl: random number routines.<br><a href="file:///smore/Users/stjohn/public_html/teaching/compbio/quiz5.html">      Quiz 5</a> (Basic Perl) due.<br>      <font>				<font color="#000000"><a href="file:///smore/Users/stjohn/public_html/teaching/compbio/project/index.html">Project:</a></font>			<font color="#009900">Target Date for Initial Design.</font>		</font><br></td>		<td><font>Moret: &nbsp;Experimental Algorithmics (see frontpage for download information)</font></td>		<td>Lecture notes on <a href="classNotes301003.pdf">computer simulations and experiments</a>,<br>      <a href="http://examples.oreilly.com/begperlbio/">Examples</a> from <a href="http://www.oreilly.com/catalog/begperlbio/">Beginning Perl for Bioinformatics</a> by James Tisdall.&nbsp;<font>&nbsp;</font></td> 	</tr><tr> <td><b><font>4&nbsp;November<br></font></b><b><font>7:45-<font color="#ff0000">9:45</font>pm</font></b><br></td>				<td align="left" valign="middle"><font>			</font>Protein Secondary Structure<br></td>		<td align="left" valign="middle">Chapters 9, 10, <br>      <font><a href="http://www-ab.informatik.uni-tuebingen.de/people/huson/welcome_en.html">Prof. Huson's</a> notes on <a href="http://www-ab.informatik.uni-tuebingen.de/teaching/ws02/script/protstruct2d.ps.gz">Protein Secondary Structure</a></font></td>		<td align="left" valign="middle"><a href="http://www.stanford.edu/group/pandegroup/folding/education/">Overview</a> from Folding@Home</td>	</tr><tr> <td><b><font>6&nbsp;November<br></font></b><b><font>7:45-<font color="#ff0000">9:45</font>pm</font></b><br></td>				<td align="left" valign="middle">Protein Tertiary Structure<br></td>		<td align="left" valign="middle">Chapters 9, 10,<br>      <font><a href="http://www-ab.informatik.uni-tuebingen.de/people/huson/welcome_en.html">Prof. Huson's</a> notes on <a href="http://www-ab.informatik.uni-tuebingen.de/teaching/ws02/script/protstruct3d.ps.gz">Protein Tertiary Structure</a></font><br></td>		<td align="left" valign="middle"><font>&nbsp;</font></td> 	</tr><tr> <td><b><font>11 November<br>7:45-9:25pm<br></font></b></td>				<td>DNA Arrays (microarrays)<br>Target Date for Programming Assignment 2.<br><font>			</font></td>		<td>      <font><a href="http://www-ab.informatik.uni-tuebingen.de/people/huson/welcome_en.html">Prof. Huson's</a> notes on <a href="http://www-ab.informatik.uni-tuebingen.de/teaching/ws02/script/protstruct3d.ps.gz">DNA Arrays</a></font><br>Chapter 11</td>	        <td><br> </td>	</tr><tr> <td><b><font>13 November</font></b><b><font><br>7:45-9:25pm<br></font></b></td>				<td><font>More on DNA Array, Haplotypes.</font><br>      <a href="quiz6.html">Quiz 6</a> (Mult. Seq. Align.) due.<br></td>		<td><font><a href="http://www-ab.informatik.uni-tuebingen.de/people/huson/welcome_en.html">Prof. Huson's</a> notes on <a href="http://www-ab.informatik.uni-tuebingen.de/teaching/ws02/script/protstruct3d.ps.gz">DNA Arrays</a></font><font>&nbsp;</font></td>		<td><br></td> 	</tr><tr>      <td valign="top"><b><font>18 November</font></b><b><font><br>7:45-9:25pm<br></font></b></td>            <td valign="middle"><font>Sequence Assembly</font>      </td>      <td valign="middle">Chapter 11      </td>      <td valign="middle"><br>      </td>    </tr>    <tr>      <td valign="top"><b><font>20 November</font></b><b><font><br>7:45-9:25pm<br></font></b></td>            <td valign="middle"><font>More on Sequence Assembly.<br></font>      </td>      <td valign="middle">Chapter 11<br>      </td>      <td valign="middle"><br>      </td>    </tr>    <tr>      <td valign="top"><b><font>25 November</font></b><b><font><br>7:45-9:25pm<br></font></b></td>            <td valign="middle"><font>Student Project Presentations. </font>Target Date for Programming Assignment 3.</td>      <td valign="middle"><br>      </td>      <td valign="middle"><br>      </td>    </tr>    <tr>      <td valign="top"><b><font>27 November</font></b></td>            <td valign="middle"><font color="#ff0000">Thanksgiving Break: No classes</font></td>      <td valign="middle"><br>      </td>      <td valign="middle"><br>      </td>    </tr><tr> <td><b><font>2 December</font></b><b><font><br>7:45-9:25pm<br></font></b></td>				<td>Student Project Presentations.<br></td>		<td><br></td>		<td><a href="ullmanSlidesF00/slides13.ps"><font>		    </font></a><br></td>	</tr><tr> <td><b><font>4&nbsp;December</font></b><b><font><br>7:45-9:25pm<br></font></b></td>				<td><font>	   	</font><font color="#ff0000">No Class.</font><br></td>		<td><font>&nbsp;</font></td> 		<td><font>&nbsp;</font></td> 	</tr><tr> <td><b><font>9&nbsp;December</font></b><b><font><br>7:45-9:25pm<br></font></b></td>				<td>Student Project Presentations.<br></td>		<td><br></td>		<td><font>		    </font><br></td>	</tr><tr> <td><b><font>11 December</font></b><b><font><br>7:45-9:25pm<br></font></b></td>				<td><font>Poster Session</font><br>Final version of project due.<br></td>		<td><font>&nbsp;</font></td> 		<td><font>&nbsp;</font></td> 		</tr><tr>      <td valign="top"><b><font>16 December</font></b></td>      <td valign="top"><font color="#ff0000">Reading Day-- No Class.</font><br>      </td>      <td valign="top"><br>      </td>      <td valign="top"><br>      </td>    </tr>    <tr>      <td valign="top"><b><font>18 December</font></b></td>      <td valign="top">All quizzes must be completed.<br>      </td>      <td valign="top"><br>      </td>      <td valign="top"><br>      </td>    </tr></tbody></table><br><br><br><small>(Last Updated: 20 November 2003.)</small><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br></body></html>