<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"><html><head><title>Presentation Schedule</title><meta http-equiv="content-type" content="text/html; charset=ISO-8859-1"></head><body><h2>Presentation Schedule<br></h2><h2> CMP 464/788: Introduction to Computational Biology<br>Lehman College, City University of New York<br>Fall 2003</h2><h2>Sequences&nbsp;</h2><h3>Tuesday, 25 November</h3>General Sequence Alignment and Scoring Matrices:<br><ul>  <li>PATTEN,&nbsp;BYRON: &nbsp;Scoring matrices <br clear="all">    <i>Abstract: A Scoring Matrix estimates the rate at which each possibleresidue in a sequence changes to each other residue over time. They are generatedusing statistics derive from known protein families. &nbsp;This presentationwill cover how they're derived and used.</i><br clear="all">    <i> &nbsp;&nbsp;</i></li>  <li>EPHRAIM,&nbsp;KIP: &nbsp;Applications of Dynamic programming <br clear="all">    <i>Abstact: Generally in sequence alignment, the idea of finding theoptimal alignment of two sequences is never found by just looking at themand coming up with an alignment with the closest match or one with very fewgaps.&nbsp; It is found by using an algorithm that involves a score to comeup with the best alignment.&nbsp; In most cases thus far, we have discussedglobal alignment of sequences where the both sequences are known and we strivefor the best alignment of the entirety of both sequences.&nbsp; However,in reality we often do not know both sequences, but we generally work withone and would be searching for an alignment of this with another sequencefound in a database or even an alignment with a portion of a very long sequence.&nbsp;This approach is called a local sequence alignment and it uses a versionof the dynamic programming algorithm developed by Dr. Temple Smith and Dr.Michael Waterman.</i><br>  </li></ul>Multiple Sequence Alignment Tools:<br><ul>  <li>MAHINDRU,&nbsp;RUCHI:&nbsp; Sequence Alignment: ClustalW <br clear="all">    <i>Abstract: Multiple sequence alignment is a critical tool in biomolecular sequence analysis.<br>Among many others, CLUSTALW is a general purpose multiple sequence alignment<br>software package for aligning DNA or proteins sequences. Mainly in this<br>tool three steps are taken to do the multiple sequence alignment. At each stage<br>various suggested algorithms are available to be used to carry out the final alignment.<br>I will present such algorithms implemented in CLUSTALW.</i><br clear="all">    <i> &nbsp;&nbsp;</i></li>  <li>TRAN,&nbsp;VIVIAN: &nbsp;Multiple Sequence Alignment: DIALIGN <br clear="all">    <i>Abstract: A method of alignment composes by collecting gap-free pairs of similar segments.No gap penalties, an avoidance of the well-known difficulty decision concerningthe choice of appropriate gap penalties. Capable of local/global, DNA/proteinand pairwise/multiple sequence alignment.</i><br clear="all">    <i> &nbsp;</i></li>  <li>BUI,&nbsp;TAM: &nbsp;BLAST &nbsp;<br clear="all">    <i>(No presentation)</i></li></ul>Hidden Markov Models and Sequence Alignment:<br><ul>        <li>CONG,&nbsp;LEI: &nbsp;Sequence alignment</li>  <li>HE,&nbsp;ZHI: &nbsp;Sequence Alignment-- HMMs</li>  <li>PATEL,&nbsp;DIPESH:&nbsp; HMMs and sequence alignment</li>  </ul><h2>Trees</h2><h3>Tuesday, 2 December</h3>Phylogenetic Tree Reconstruction Techniques:<br><ul>  <li>HASANGJEKAJ,&nbsp;JAH &amp; LOU,&nbsp;QIAN (JANE):&nbsp; Phylogenetics: Dyadic Closure Methods<br clear="all">    <i>Abstract: A phylogenetic tree is a leaf-labeled, binary tree whichtracks the genetic similarities of a set of closely related species. Sinceall currently available tree reconstruction algorithms build unrooted trees,we&#8217;d also follow the same principles. We will discuss how long the sequencesshould be in order to reconstruct the tree. To analyze this, we will referto a tree reconstruction method called Dyadic Closure Method that generatesa phylogenetic tree in O(n2k + n5 log(n) ), where n is the number of sequencesand k is the sequence length. &nbsp;</i><br>    <br>  </li>    <li>REYES,&nbsp;JUAN:&nbsp; phylogenetic methods-- quartets &nbsp;<br clear="all">    <i>(No presentation)</i><br clear="all"> &nbsp;</li>    <li>WALSH,&nbsp;RACQUEL: &nbsp;Neighbor-Joining Algorithm: Proof and Examples <br clear="all">    <i>Abstract: My Intent is to understand and explain the neighbor-joiningalgorithm (NJ) of Saitou and Nei (1987). It is&nbsp; proposed for the&nbsp;reconstructing of phylogenetic trees from evolutionary distance data. Theprinciple of this method is to find pairs of taxonomic units (neighbors)that minimize the total branch length at each stage of clustering of startingwith a star like tree. The branch lengths as well as the topology of a parsimonioustree can quickly be obtained by using this method. Neighbor-Joining has arunning time of O(n^2).</i><br clear="all">    <i> &nbsp;</i></li><li>FLORENCIO,&nbsp;DAVID: &nbsp;Searching Tree Space: MrBayes</li></ul>Visualization and Comparison Tools:<br><ul>  <li>AVILA,&nbsp;PEDRO &amp; CORLETTO,&nbsp;PETER: &nbsp;Comparison of Sequence Alignment Visualization Tools<br clear="all">    <i>Abstract: We are presenting different types of visualization applicationsthat can be used to display protein and nucleotide sequence alignments. Theseapplications shade the regions of similarity with beautiful colors for easyidentification of such regions. We will show the advantages and disadvantagesof using the following two types of applications: a) applications that wrapthe alignment around to the next line, and b) applications that display anentire sequence alignment on a single line.<br>&nbsp;A) Applications that wrap around to the next line: These applicationshave a limit on the number of characters or bases displayed in a single line.If any given alignment is longer than this limit, the alignment continueson the next paragraph. As our case of studies, we will use two applicationscalled GeneDoc and SeaView.<br>&nbsp;B) Applications that display an entire sequence alignment on a singleline: for these applications, there is no limit on the number of charactersor bases that can be displayed on a single line. These applications displayan entire sequence alignment on a single line, shading regions of similaritywith a specific color. For a detailed inspection of a region of similarity,the user can zoom into a region of interest and then examine the sequencescharacter by character. In our presentation, we will use two applicationsthat visualize sequences using this approach: Artemis and LALNVIEW.<br>Pedro Avila will compare and contrast GeneDoc and Artemis. &nbsp;Peter Corletto will compare and contrast SeaView and LALNVIEW. </i><br clear="all">&nbsp;&nbsp;</li>  <li>MARTINEZ,&nbsp;SONIA: &nbsp;An Interactive Approach to DNA Sequence Alignments based on Phylogenetic Relationships (Phylo-VISTA)<br>    <i>Abstract: Phylo-VISTA is an interactive multiple DNA sequence alignmenttool which measures the DNA sequences of several species while consideringtheir phylogenic relationship. The results (aligned sequences) are visuallypresented to the user which makes it easier to compare the similarities ordifferences in the sequences between species. These sequences can determinetheir evolutionary relationships. Phylo-VISTA determines the results by followingspecific tasks. These tasks include, display of multiple alignments basedon their corresponding phylogenetic tree; determined similarity of nodesin the tree displayed on a user window; visual display of sequence conservationby a line plot; and presentation of relative statistics with observations.The Phylo-VISTA tool is easily accessible online as a Java Applet at http://www-gsd.lbl.gov/phylovista/.</i><br>    <br></li>  <li>TINTA,&nbsp;SANDRA: &nbsp;Visualizing Trees: H3<br>    <i>Abstract: There are many research efforts to implement tools to visualizetress, but most tools can handle limited datasets with few thousand nodes.&nbsp;H3 can handle over 100,000 edges with minimal visual cluster.&nbsp; H3 provideslayout and interactive navigation of node-liked graph in 3D hyperbolic spacewhich allows the user to see a great deal of the context around the focusnode.&nbsp; Even though, H3 is a great tool to visualize large trees, H3is not useful when attempting to compare trees.</i><br clear="all"> &nbsp;&nbsp;</li>  <li>PENA,&nbsp;RONNY:&nbsp; &nbsp;Felsenstein's distance in treeComp</li>    </ul><h2>Proteomics &amp; Genomics</h2><h3>Tuesday, 9 December</h3>Due to lack of time on 2 December, Sandra Tinta and Ronny Pena will be present first (abstracts above), followed<br>by:<br><br>Protein Folding and Structure:<br><ul>  <li>BUGA,&nbsp;NATALYA: Predicting Protein Function and Structure from Sequence Using the PROSPECT Software<br>    <i>Abstract: One of the most successful tools for predicting proteinstructure by sequence presented at CASP IV competition was the PROSPECT softwarepackage developed at Oak Ridge National Laboratories.&nbsp; The results shownby the PROSPECT system placed the group presenting this software in the top4% of the highest scoring groups in CASP.<br>    <br>In my presentation, I will briefly define the problem of predicting the proteinstructure from sequence and possible approaches used to solve this problem.&nbsp;Also, the presentation will describe the goals and achievements ofthe CASP competition.&nbsp; The project will concentrate on describing the features,underlying techniques, and algorithms used by the PROSPECT software systemand demonstrate the results produced by the tool. </i><br clear="all">    <i>&nbsp; &nbsp;</i></li>    <li>LOWDEN,&nbsp;CECIL:&nbsp; Proteins<br clear="all"> &nbsp;</li>  <li>NGNOSSE,&nbsp;SEVERIN:&nbsp; Fast Algorithms for Determining Protein Structure Similarity &nbsp;<br clear="all">    <i>Abstract: &nbsp;This presentation is based on published work of SomenathBiswa (India Institute of Technology Kanpur) and Samarjit Chakraborty(EiggenossischeTechnische Hoschschule Zurich).<br>    <br>Identifying the 3D structural similarity that exist between 2 proteins hasbecome a challenge inthe scientific comunity. Many techniques have been proposedand among them is the one that&nbsp; use a new metric called the bottleneckmatching metric. &nbsp;Using this measure, all the algorithms presented sofar suffer from running time which are high degree polynomials in the numberof atoms in the protein molecules, making them infeasible for practical applications.<br>This paper shows that by exploiting a very simple structural property ofproteins , the running time of some of&nbsp;these algorithms can be considerabyimproved. Combining this additional information with some standard algorithmstechniques such as randomization, and/or approximate matching scheme forbipartite grahs, the new algorithms so obtained have running times whichare near linear.&nbsp;</i> <br>    <br></li><li>SINGUINAM,&nbsp;PALILBA: &nbsp;Protein folding and deseases.<br>    <i>Abstract: The human body makes at least 50,000 different proteinsand possibly twice that many. They are the essential working parts of livingmatter. If a cell is thought of as a house, then proteins are just abouteverything in it. They are the furniture, the fixtures, the lumber and soon, but what happen if one of this contains doesn&#8217;t work? Well my presentationwill aim on how proteins fold and what kind of diseases they may cause ifwe get some misfolding proteins in the cell. </i><br>    <br></li>  <li>STEWART,&nbsp;ANDREW: &nbsp;Proteomics: EXPASY suite of tools</li></ul>Gene Detection and Microarrays:<br><ul>  <li>WANG,&nbsp;HONG:&nbsp; DNA Microarray Technology and Clustering Algorithms<br>  <i>Abstract: Microarray technology&nbsp; using cDNAs&nbsp; make possible the measurementof levels and patterns of cell and tissue-specific gene expression. Cy3 andCy5 RNA samples are hybridized to the chip containing thousands of specificgenes, analyzing scanned microarray images. Agglomerative hierarchical clusteringis a kind of algorithms about microarray, it is simple and can be used fairlysuccessfully for analysis of microarray data.</i><br clear="all">    <i> &nbsp;&nbsp;</i></li><li>OLIVO,&nbsp;ALVIN: &nbsp;Microarrays and detection of disease<br clear="all"> &nbsp;&nbsp;</li><li>XU,&nbsp;MENG and DENG, YUANGUANG: Clustering of Gene Expression Data in Microarray<br>    <i>Abstract: This group project is focused on Introduction to Microarray,cluster algorithm, and also try to understand some new research paper onMicroarray reseach. &nbsp;First, we will search for some background knowledgeof Microarray experiments, including basic biology and genetics resources.&nbsp;Secondly, according to the not made in Bioimformatics Class, understandtwo main approaches to solve the data clustering, supervised Vs. unsupervised.&nbsp;Finally, we will find a new research paper to try use what we learnfrom class and research of this project to study, understand its algorithmand experiment outcome.</i><br>    <br></li>  </ul>Genomics and DNA Computing<br><ul>  <li>PHAM,&nbsp;KHANH: &nbsp;Genomics: Pipmaker <br>    <i>Abstract: In this project, I discuss about an automated web basedserver for generating alignments pips (percent identity plot). A pip showsthe position in one sequence of each aligning gap-free segment and plotsits percent identity. I also provide a plot of the position of each aligningsegment in both species. I refer to these as dot plots, even though matchesshown in conventional dot plots need not be contained within a statisticallysignificant alignment and those in our plots are. Both displays allow richannotation to be plotted along the appropriate axes to aid in correlatingaligning segments with functional or structural features of the sequence.I provide examples of the application of PipMaker for finding exons and candidateregulatory elements in human and mouse. The pipmaker web based server isable to compare a sequence from a human with a mouse sequence.</i><br clear="all">    <i> &nbsp;</i></li><li>SINGH, TYRON: Data Mining in Bioinformatics <br clear="all">&nbsp;   </li>    <li>MATSUO,&nbsp;KOYA:&nbsp; DNA Computing -- a survey<br>    <i>Abstract:&nbsp; This is a survey into the topic of DNA computing,with overviews in its methodologies and technologies, principles, a bit ofhistory, and its position within computational biology. How DNA computinghas shed light<br>on the major topics of computer science, such as NP class problems, and howit has the possibility to influence computational biology, as well as thewhole notions of computation and computer. It can give different perspectiveson string operational algorithms, sequence alignments, and technologies behindcomputational biology, and I'd like to show this excitement surrounding thisconvergence between the two disciplines, with a vision of future computer.</i></li></ul>&nbsp;  <br><br><hr width="100%" size="2"><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br>&nbsp;&nbsp; &nbsp;<br><br><br><br><br><br><br><br><br><br><br><br></body></html>