<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"><html><head><title>CMP 464/788, Lehman College, Fall 2003 Program 4</title><meta http-equiv="content-type" content="text/html; charset=ISO-8859-1"></head><body><h1>Programming Assignment 4<br></h1><h2> CMP 464/788: Introduction to Computational Biology<br>Lehman College, City University of New York<br>Fall 2003</h2><hr><font color="red">The target date for this programming assignment is Tuesday, 9 December.</font><br><hr><b>Topics:</b> More on Perl Basics: Hashes<h2>General Information</h2>The following programs should be written in the perl programming language and turned in via the <b>Dropbox</b> option in Blackboard. &nbsp;When turning in your programs, you should submit <b>plain text files</b> and call the programs <tt>youlastnameProgramX.pl</tt> where X is the program number. &nbsp;<br><br>To obtain full credit for your program, the initial comment block at thebeginning of the file should include your name, your preferred email, anda short description of the program. &nbsp;You should comment any non-obvioussections of codes. &nbsp;Use your own words to describe the program in comments.&nbsp;Remember that copying others programs is considered cheating. &nbsp;Submittedprograms will be randomly compared to look for similarity and cheating.<h3>Undergraduate Programs</h3>The following program are for those taking the course for undergraduate credit:<br><ul>  <li><b>UG7:</b> &nbsp;Write a perl program that takes a list of genes,sequences and their assession numbers from GenBank and formats them so theycan be easily read into a hash (which you do in the next program). &nbsp;Perlis used extensively in bioinformatics for converting data that was generatedin one format to another. &nbsp;For example, a common question is check alist of sequences to check if it's in FASTA format, and from there choosethe appropriate program to run. Write a subroutine that checks an arrayof data and returns true if it's in FASTA format (see for details of the format)<font color="#ff0000"> </font>&nbsp;Your program should read in a set of sequences from standard input and print out the answer.<i><br>    </i></li>  <li><b>UG8:</b> &nbsp;Write a perl program that searchesby name for a gene in a hash, and returns its assession number and sequence.Your program should first prompt the user for the name of the gene file (assumethat the input file is produced by your program UG7 and, as such, is in aspecific form). &nbsp;Your program should then repeatly ask the user forthe name of gene and print out the information, until the user enters a blankline to quit. &nbsp;<i><br>    </i></li></ul><br><h3>Graduate Program</h3><b>In addition</b> to the undergraduate programs, those taking the course for graduate question should also do the following program:<br><br><ul><li><b>G4:</b> &nbsp;Write a perl program that takes a tree in Newick formatand converts it to a NEXUS file (a common file format). &nbsp;DESCRIBE NEXUS...&nbsp; to see if the input is a tree in Newick format and prints the answerto the screen. &nbsp;You may assume that the input will represent unweightedtrees (ie you do not need to consider weights when checking if it's in theright format) and use the subroutine you wrote for Program G3. &nbsp;Forexample, if the input to your program is (t1,(t2,t3)), A few minutes of familiarizingyourself with the format ahead of time will same much time in the programming.&nbsp;For parsing the input, you might find the push() and pop() functionsapplied to a list helpful.&nbsp;</li></ul><br><br><br></body></html>